docs/annotations.md

```diff
@@ -7,6 +7,7 @@ Contour supports the following annotations.
 
  - `kubernetes.io/ingress.class`: The Ingress class that should interpret and serve the Ingress. If not set, then all Ingress controllers serve the Ingress. If specified as `kubernetes.io/ingress.class: contour`, then Contour serves the Ingress. If any other value, Contour ignores the Ingress definition. You can override the default class `contour` with the `--ingress-class-name` flag at runtime. This can be useful while you are migrating from another controller, or if you need multiple instances of Contour.
  - `ingress.kubernetes.io/force-ssl-redirect`: Requires TLS/SSL for the Ingress to Envoy by setting the [Envoy virtual host option require_tls](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route.proto.html#envoy-api-field-route-virtualhost-require-tls)
+ - `ingress.kubernetes.io/rewrite-target`: [Target prefix rewrite](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route.proto#envoy-api-field-route-routeaction-prefix-rewrite).
```

Comment:
I've not been able to find a reference to anyone else using this annotation, so I think it would have to be `contour.heptio.com/rewrite-target`

I'm not sure how this is supposed to work. Looking at the envoy docs this field applies per route, not per virtualhost, so by setting this on the ingress object we would rewrite every path to be "/". So, yeah, I don't really understand how this is supposed to work. 

Perhaps if it was `contour.heptio.com/rewrite-target: "/foo=/bar,/testing=/prod"` would make more sense as we could parse that string, and if the route is `/foo` then apply the rewrite rule on the generated route.

Reply:
@davecheney I personally think the syntax listed on standard k8s documentation indicates one rule applies all, see Types of Ingress -> Simple fanout:
https://kubernetes.io/docs/concepts/services-networking/ingress/#types-of-ingress

Also in traefik annotation (traefik.ingress.kubernetes.io/rewrite-target: /users):
https://docs.traefik.io/configuration/backends/kubernetes/#annotations

Also AWS Application Load Balancer Ingress Controller:
https://github.com/kubernetes-sigs/aws-alb-ingress-controller/issues/142#issuecomment-316074859

Voyager:
https://github.com/appscode/voyager/blob/70a3b3ed88b2bca88fb220f9273fae7fc0f57bbd/docs/examples/certificate/google-cloud/ing-tls.yaml
https://github.com/appscode/voyager/blob/70a3b3ed88b2bca88fb220f9273fae7fc0f57bbd/docs/examples/certificate/route53/ing-tls.yaml
https://github.com/appscode/voyager/blob/beea9abe7e4ab411950d2706f2d54e0014b1fa84/hack/example/ingress.yaml

Rest of them have their own specific implementations. Just my thought.